### PR TITLE
[WIP] Content/uploader development

### PR DESCRIPTION
**Pull request by nbumbarger**
I know a handful of people who've had trouble starting work on the OAM uploader, including myself, because of uncertainty over how the components fit together to get the system up and running. Since what I wanted to write addresses the interconnectedness of the components and also some aspects of getting started, I thought it would be a better fit for here in the docs than in any of the individual READMEs.

### CODE REVIEW

File: docs/uploader/34_development.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+- It will be necessary to use [nvm](https://github.com/creationix/nvm) in development- the two forms must be built use node 4.x, while the API must be built using node 0.12.
+
+- When launching the API, several environment variables must be set using the following command:
+
+  ```
+  GDAL_TRANSLATE_BIN={your GDAL 2 path} ADMIN_USERNAME={your choice} ADMIN_PASSWORD={your choice} AWS_SECRET_KEY_ID={} AWS_SECRET_ACCESS_KEY={} AWS_REGION=us-east-1 OIN_BUCKET=hotosm-oam-staging DBURI={} DBURI_TEST=mongodb://localhost/oam-uploader-test node index.js
+  ```
+
+  The script does not fully inherit the variables in the local.env file, so this override is necessary even if it has been populated. Contact a maintainer for the full AWS credentials and database paths.
```
Comment (danielfdsilva) [inline docs/uploader/34_development.md]: You can do `source local.env` before starting the api to load all the variables.
Also, variables set from ENV, will always override the ones in the config files.

**Review by danielfdsilva — CHANGES_REQUESTED** [review]

**Review by danielfdsilva — APPROVED** [review]

**Comment by kamicut** [comment]
@danielfdsilva if you can look at the styling that would be great. What do you think of the wording in this version of the ecosystem docs?

**Comment by danielfdsilva** [comment]
@kamicut overview looks nice! I think it captures the system well.
I'd say the next thing to do is ensure that the README's of the different components have good instructions on how to be setup (i.e. what are the dependencies they have. For example the uploader would need the uploader-api running).
We could also setup a "development" subpage where we list everything that's needed to setup each "group". I kind of like this approach. Thoughts?

**Comment by kamicut** [comment]
@danielfdsilva 
1. Is this branch ready to merge? We can open another one for the other development pages. 
2. I think we can have a development subpage, but it should just point to the development subpages of each component. It could be more readable to divide the current ecosystem doc into "Intro" "Catalog Group" "Uploader Group" "Run your own OAM"

**Review by smit1678 — CHANGES_REQUESTED** [review]
First glance, looks good. Fixed the conflict. Can we complete the Roll Your Own section a bit more?

**Comment by kamicut** [comment]
@smit1678 I'm wondering how much should be in the "Roll Your Own" section, and how much should be in the individual READMEs. 

My thinking was that we could direct developers to the "development" page in the uploader and catalog sections, which would lead to the READMEs. But that could be a lot of indirection. (e.g: `Roll Your Own -> Uploader Development -> oam-uploader-admin` README)

Another option would be to have all the links in the github READMEs from Roll Your Own, skipping the Uploader Development OAM docs.

What do you think should be in the Roll Your Own section?

**Comment by smit1678** [comment]
@kamicut Yes, agreed that we should push towards the READMEs for more detailed setup and configuration for each component. What about just merging Roll Your Own and Development pages?

**Comment by kamicut** [comment]
@smit1678 I moved the development docs into Roll Your Own, and linked to the READMEs. I have also gone over the individual READMEs and updated anything that is missing for local dev. Let me know if this is enough for ecosystem + local development documentation.

**Comment by danielfdsilva** [comment]
@smit1678 can we get a final set of eyes on this one to merge?

**Comment by smit1678** [comment]
👍  I'm going to merge this and break out separate issues to follow up on a couple ways to improve content based on some feedback.

**Review by smit1678 — APPROVED** [review]

**Pull request merged**